apps/api/src/prisma/prisma.repository.ts

```diff
@@ -155,6 +156,51 @@ export class PrismaRepository {
     })
   }
 
+  /**
+   * Invalidate Old OTPs for a User
+   *
+   * This method invalidates old OTPs (One-Time Passwords) associated with a user.
+   * It finds and deletes OTPs that belong to the user and have not expired yet.
+   *
+   * @param email - The email address of the user for whom old OTPs should be invalidated.
+   *
+   * @example
+   * ```typescript
+   * await invalidateOldOtps('user@example.com');
+   * ```
+   */
+  private async invalidateOldOtps(email: string): Promise<void> {
+    const user = await this.prisma.user.findUnique({
+      where: {
+        email
+      }
+    })
+    if (user) {
+      const oldOtpCodesToDelete = await this.prisma.otp
```

Comment:
I think you can spare yourself (and us!) some time complexity, and pass the conditions directly in `prisma.otp.deleteMany` call.

Reply:
I have updated the changes, please review it @rajdip-b